### PR TITLE
Barry crud

### PR DESCRIPTION
**Pull request by BarryShovlin**
##Changes
-Can now delete a category that is not tied to a post.

##Check
-that the new method in Post controller and sql statement in post repository are functioning.
-that categories are deleted if not tied to a post.
-that the window alert shows up stating that a category may not be deleted if tied to a post.

### CODE REVIEW

**Comment by MegSemrad** [comment]
👍

File: Tabloid/client/src/providers/CategoryProvider.js [hunk 1]
```diff
@@ -49,4 +64,16 @@ export const CategoryProvider = (props) => {
         </CategoryContext.Provider>
     );
 };
-export default CategoryProvider;
\ No newline at end of file
+export default CategoryProvider;
+
+const getAllCommentsByPostId = (postId) => {
```
Comment (rickiele) [inline Tabloid/client/src/providers/CategoryProvider.js:69]: Hey I don't think this is supposed to be here!

Comment (rickiele) [reply Tabloid/client/src/providers/CategoryProvider.js:69]: ./src/providers/CategoryProvider.js
  Line 70:12:  'getToken' is not defined     no-undef
  Line 78:19:  'setComments' is not defined  no-undef

**Comment by MegSemrad** [comment]
👍

**Comment by rickiele** [comment]
👍

**Pull request merged**